### PR TITLE
etcd-tester: fix compact timeout

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5440.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @heyitsanthony @xiang90 Sorry for multiple changes. The functional-tester has a lot of bugs for the last few weeks. The major change is adjust the compact request timeout by the number of entries. 

Confirmed that it fixes https://github.com/coreos/etcd/issues/5440.

Won't deploy this because I am still seeing https://github.com/coreos/etcd/issues/5317.

Please review.

Thanks.

**Comment by gyuho** [comment]
/cc @xiang90 PTAL. Just kept related changes for the issue.

Thanks.

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -105,7 +106,14 @@ func (tt *tester) runLoop() {
 		}
 
 		revToCompact := max(0, tt.currentRevision-10000)
-		if err := tt.compact(revToCompact); err != nil {
+		compactN := revToCompact - prevRevToCompact
+		timeout := time.Minute
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: this jitter is too long. probably 10 second.

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -105,7 +106,14 @@ func (tt *tester) runLoop() {
...
+		if prevRevToCompact != 0 && compactN > 0 {
+			timeout += time.Duration(compactN/60000) * time.Second // 60K-entry compaction takes about 1-second
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: let's say 50000 to be safe. also we should make this a const.

File: tools/functional-tester/etcd-tester/tester.go [hunk 3]
```diff
@@ -41,6 +41,7 @@ func (tt *tester) runLoop() {
 		tt.status.setRound(round)
 		tt.status.setCase(-1) // -1 so that logPrefix doesn't print out 'case'
 		roundTotalCounter.Inc()
+		prevRevToCompact := tt.currentRevision
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: startRev

**Comment by xiang90** [comment]
lgtm

**Pull request merged**